lib/spack/spack/util/path.py

```diff
@@ -27,6 +28,38 @@
     'tempdir': tempfile.gettempdir(),
 }
 
+# This is intended to be longer than the part of the install path
+# spack generates from the root path we give it.  Included in the
+# estimate:
+#
+#   os-arch      ->   30
+#   compiler     ->   30
+#   package name ->   50   (longest is currently 47 characters)
+#   version      ->   20
+#   hash         ->   32
+#   buffer       ->  138
+#  ---------------------
+#   total        ->  300
+SPACK_MAX_INSTALL_PATH_LENGTH = 300
+SPACK_PATH_PADDING_CHARS = 'spack_path_placeholder'
+_spack_sys_max_path_length = None
+
+
+def get_system_path_max():
+    global _spack_sys_max_path_length
```

Comment:
you could use `@llnl.util.lang.memoized` here to simplify.

Reply:
Well that's cool, thanks for pointing it out 👍